### ISSUE TITLE
[BUG] anvi-pan-genome does not allow rerunning without repeating time-consuming aa search.

### ISSUE THREAD

**Issue by snystrom** [merenlab/anvio#1865]
## Short description of the problem

`anvi-pan-genome` won't rerun w/ cached previous results. Chatted w/ Meren about this briefly on Slack & was asked to report here.

## anvi'o version

```
Anvi'o .......................................: hope (v7.1)

Profile database .............................: 38
Contigs database .............................: 20
Pan database .................................: 15
Genome data storage ..........................: 7
Auxiliary data storage .......................: 2
Structure database ...........................: 2
Metabolic modules database ...................: 2
tRNA-seq database ............................: 2
```

## System info

Ubuntu 20.04 (Focal) (Actually, a docker container w/ this OS: `https://hub.docker.com/r/jupyter/datascience-notebook/dockerfile`).

Anvi'o was installed using the current instructions for 7.1 on the website, but using `mamba` in place of conda. Successful install was confirmed with `anvi-self-test --nointeractive`.

## Detailed description of the issue

`anvi-pan-genome` will not rerun using preexisting data as expected based on the pangenomics tutorial.

Based on this:
![image](https://user-images.githubusercontent.com/17264973/147163213-0532d2a5-aedf-4b53-a92f-e7b502b87a34.png)

I interpreted this to mean that I can rerun anvio using a new setting (`--min-occurrence 2`) if setting `--overwrite-output-destinations`. However, this does not appear to be the case.

I generated a pangenome & genome store from some test data, then can construct a pangenome:

```
# Initial run works great:
anvi-pan-genome -g MY-GENOMES.db -n My-Pan 
```

Next, I want to rerun with a new `--min-occurrence` & I get a "this dir exists" error.

```
# Subsequent run w/ new settings:
anvi-pan-genome -g MY-GENOMES.db -n My-Pan --min-occurrence 2 

WARNING
===============================================
If you publish results from this workflow, please do not forget to cite DIAMOND
(doi:10.1038/nmeth.3176), unless you use it with --use-ncbi-blast flag, and MCL
(http://micans.org/mcl/ and doi:10.1007/978-1-61779-361-5_15)
Functions found ..............................: COG20_FUNCTION, COG20_CATEGORY, COG20_PATHWAY                                                      
Genomes storage ..............................: Initialized (storage hash: hash905ae9a9)                                                           
Num genomes in storage .......................: 86
Num genomes will be used .....................: 86
Pan database .................................: An existing database,
                                                /home/snystrom/anvio-reprex/My-Pan/My-Pan.db, has
                                                been initiated.
Genomes ......................................: 87 found


Config Error: Anvi'o will not overwrite an existing pan database. Please choose a different 
              name or remove the existing database ('/home/snystrom/anvio-reprex/My-Pan/My-Pan.db') first.
```

Then if I try to force overwriting like is hinted at in the tutorial, the error persists. (Also, the `--overwrite-output-destinations` flag does not appear in the `--help` text or on the anvio documentation site?).
```
# Retry with --overwrite-output-destinations:
anvi-pan-genome -g MY-GENOMES.db -n My-Pan --min-occurrence 2 --overwrite-output-destinations

WARNING
===============================================
If you publish results from this workflow, please do not forget to cite DIAMOND
(doi:10.1038/nmeth.3176), unless you use it with --use-ncbi-blast flag, and MCL
(http://micans.org/mcl/ and doi:10.1007/978-1-61779-361-5_15)
Functions found ..............................: COG20_FUNCTION, COG20_CATEGORY, COG20_PATHWAY                                                      
Genomes storage ..............................: Initialized (storage hash: hash905ae9a9)                                                           
Num genomes in storage .......................: 86
Num genomes will be used .....................: 86
Pan database .................................: An existing database,
                                                /home/snystrom/anvio-reprex/My-Pan/My-Pan.db, has
                                                been initiated.
Genomes ......................................: 87 found


Config Error: Anvi'o will not overwrite an existing pan database. Please choose a different 
              name or remove the existing database ('/home/snystrom/anvio-reprex/My-Pan/My-Pan.db') first.
```

I'm fine rerunning w/ different settings for now in parallel jobs, but it'd be wonderful to be able to cut some runtime down for tweaking. Alternatively, I may be doing something silly, and would be overjoyed to learn so.

Thanks for your help!
   -Spencer

**Comment by meren** [comment]
Hey @snystrom,

I'm afraid we need to clarify the tutorial, because the functionality is in there, but getting to it is a bit challenging, I guess. I did the same thing, and created a pangenome:

```
anvi-pan-genome -g TEST-GENOMES.db -n My-Pan
```

Then, as you did, I tried to generate a new pangenome using an additional filter with the hope that this call would use existing files:

```
anvi-pan-genome -g TEST-GENOMES.db -n My-Pan --min-occurrence 2
```

But indeed, I got the same error you did:

```
Config Error: Anvi'o will not overwrite an existing pan database. Please choose a different
              name or remove the existing database ('/Users/meren/Downloads/PAN_TEST_FILES/My-
              Pan/My-Pan-PAN.db') first.
```

So the hidden message here is that anvi'o says it will not overwrite **an existing pan database**. So anvi'o doesn't actually have a problem with the directory. The name of the database is populated from `-n My-Pan`, which defines not only the database path, but also the output directory path when none is declared explicitly. So if we could use the same directory with a *different* database path, our problem would have been resolved.

So when I run this one instead:

```
anvi-pan-genome -g TEST-GENOMES.db -n My-Pan-2 -o My-Pan/ --min-occurrence 2
```

Anvi'o uses the existing output files as I can see the following notes in the output messages:

```
(...)

WARNING
===============================================
Notice: A diamond database is found in the output directory, and will be used!

WARNING
===============================================
Notice: A DIAMOND search result is found in the output directory: skipping
BLASTP!

(...)
```

And at the end of this, there are two pan databases in the same output directory:

``` bash
$ ls My-Pan/*db
My-Pan/My-Pan-PAN.db
My-Pan/My-Pan2-PAN.db
```
I hope this clarifies.

Thank you very much for your detailed report. I just updated the pangenomics tutorial with https://github.com/merenlab/web/commit/a2ff73288136a759dd07bc18634e61a07a2e8f4b so it has a link to this issue in case it helps others, as well.


Best wishes,
